machinery/topology_test.go

```diff
@@ -369,3 +397,111 @@ func TestTopologyWithRuntimeObjects(t *testing.T) {
 
 	SaveToOutputDir(t, topology.ToDot(), "../tests/out", ".dot")
 }
+func TestTopologyHasLoops(t *testing.T) {
+	apples := []*Apple{{Name: "apple-1"}, {Name: "apple-3"}}
+	oranges := []*Orange{
+		{Name: "orange-1", AppleParents: []string{"apple-1"}},
+		{Name: "orange-3", AppleParents: []string{"apple-3", "apple-1"}},
+		{Name: "orange-4", AppleParents: []string{"apple-3"}},
```

Comment:
Same. Name the objects sequentially maybe?

```suggestion
		{Name: "orange-2", AppleParents: []string{"apple-3", "apple-1"}},
		{Name: "orange-3", AppleParents: []string{"apple-3"}},
```